library_api/entity/book.go

```diff
@@ -0,0 +1,49 @@
+package entity
+
+import (
+	"encoding/json"
+	"strconv"
+)
+
+type Book struct {
+	Name   string `json:"name"`
+	Author string `json:"author"`
+	Year   int    `json:"year"`
+}
+
+type ByYear []Book
+
+func (a ByYear) Len() int           { return len(a) }
+func (a ByYear) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a ByYear) Less(i, j int) bool { return a[i].Year < a[j].Year }
+
+func (b *Book) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
+		Name   string `json:"name"`
+		Author string `json:"author"`
+		Year   string `json:"year"`
+	}{
+		Name:   b.Name,
+		Author: b.Author,
+		Year:   strconv.Itoa(b.Year),
+	})
+}
+
+func (b *Book) UnmarshalJSON(data []byte) error {
+	book := &struct {
+		Name   string `json:"name"`
+		Author string `json:"author"`
+		Year   string `json:"year"`
+	}{}
```

Comment:
you can predefine and reuse the struct
let's name it `BookRequestAndResponse` and just reuse it every time you handle post+get requests